src/runtime/audience-action-flow.js

```diff
@@ -266,12 +266,38 @@ export class AudienceActionFlow {
   // eslint-disable-next-line no-unused-vars
   handleSurveyDataTransferRequest_(request) {
     // @TODO(justinchou): execute callback with setOnInterventionComplete
-    // and Google Analytics, then check for success
+    // then check for success
+    const gaLoggingSuccess = this.logSurveyDataToGoogleAnalytics(request);
     const surveyDataTransferResponse = new SurveyDataTransferResponse();
-    surveyDataTransferResponse.setSuccess(true);
+    surveyDataTransferResponse.setSuccess(gaLoggingSuccess);
     this.activityIframeView_.execute(surveyDataTransferResponse);
   }
 
+  /**
+   * Logs SurveyDataTransferRequest to Google Analytics. Returns boolean
+   * for whether or not logging was successful.
+   * @param {SurveyDataTransferRequest} request
+   * @return {boolean}
+   * @private
+   */
+  logSurveyDataToGoogleAnalytics(request) {
+    const gtag = this.deps_.win().gtag || null;
+    if (gtag && typeof gtag !== 'function') {
+      return false;
+    }
+    request.getSurveyQuestionsList().map((question) => {
+      const answer = question.getSurveyAnswersList()[0];
+      gtag('event', 'survey submission', {
```

Comment:
Do we want to put this event message into https://github.com/subscriptions-project/swg-js/blob/756d962e748475a9f6f5f95c7fe376709ed613a1/src/runtime/event-type-mapping.js#L126-L218?

Reply:
No, I think your approach is right that the handler directly invokes `gtag` function if available, rather than delegating it to event manager, so that we have control over success/failure of the data transfer.